### DISCUSSION TITLE
Submit button does not invoke the route

### DISCUSSION THREAD

**Discussion by ramontvr** [adonisjs/core#1289] [Help]
hello all
Im trying to do a request (a query) but the button just refresh the screen.
I cant find the error, i used the same code in another project. I already created another file but no success.
I cant debug with consle.log. its like my button werent working when I press

My View
```
@layout('layout.app')
@section('content')

<form action="/buscarpordata" method="get">

<input type="date" name="datanota">
<input type="submit" value="Buscar">

</form>

@endsection
```

Im trying to call a controller with this route:
```
Route.on('/buscarpordata').render('buscanotapordata').middleware(['auth'])
Route.get('/buscarpordata', 'BuscaPorDController.search').middleware(['auth'])
```

My Controller
```
'use strict'

class BuscaPorDController {


 search({request, response, session, view}) {
   let query = request.input('datanota')
    const notas = await Database
.select('notas.id', 'numero', 'notas.created_at', 'fornecedors.nome', 'fornecedors.cnpj', 'username')
.from('notas')
.where('notas.created_at','like', '%'+query+'%')
.join('fornecedors', 'fornecedors.id', 'id_fornecedor_fk')
.join('users', 'users.id', 'id_usuario_fk')

if(notas) {
    console.log(notas)
}
    
}

}
module.exports = BuscaPorDController
```

**Comment by RomainLanz** [comment]
Hey @ramontvr! 👋 

Your form should be sent with the `POST` method, you are sending it with `GET`.
Note that you should also change the route declaration to handle `post` and not `get`.

```ts
       👇🏼
Route.post('/buscarpordata', 'BuscaPorDController.search').middleware(['auth'])
```

**Reply by ramontvr** [reply]
Hi @RomainLanz  thanks for helping 
why Post? Now im a little bit confused. Im trying to filter fields by date. Shouldnt i use GET?

**Reply by alainrc2005** [reply]
Se recomienda enviar los datos hacia el servidor utilizando POST. El método GET es utilizado mayoritariamente para traer datos o páginas web, la diferencia es como se pasan los datos en GET y POST al servidor. Con GET los datos se pasan en la URL y con POST en el cuerpo de la petición.

Mi sugerencia personal es que hagas uso de AJAX y utilices POST en las llamadas al servidor. Espero que te sirva de ayuda y utilicé español por el nombre de las variables y controlador.

**Comment by McSneaky** [comment]
HTML forms will do page refresh by default. You have to handle it manually if you want to do it in the background with ajax call without page refresh

Something like this
```html
<form onsubmit="manualHandler">
  ...inputs
</form>

<script>
function manualHandler(event) {
  // It's important to prevent default event, which is "refreshing the page"
  event.preventDefault()
  // Handle rest of it manually
}
</script>
```
